internal/services/account_sponsorship_service.go

```diff
@@ -201,12 +221,16 @@ func (s *accountSponsorshipService) WrapTransaction(ctx context.Context, tx *txn
 		return "", "", fmt.Errorf("getting transaction envelope: %w", err)
 	}
 
-	return feeBumpTxe, s.SignatureClient.NetworkPassphrase(), nil
+	return feeBumpTxe, s.DistributionAccountSignatureClient.NetworkPassphrase(), nil
 }
 
-func NewAccountSponsorshipService(signatureClient signing.SignatureClient, horizonClient horizonclient.ClientInterface, maxSponsoredBaseReserves int, baseFee int64, models *data.Models) (*accountSponsorshipService, error) {
-	if signatureClient == nil {
-		return nil, fmt.Errorf("signature client cannot be nil")
+func NewAccountSponsorshipService(distributionAccountSignatureClient, channelAccountSignatureClient signing.SignatureClient, horizonClient horizonclient.ClientInterface, maxSponsoredBaseReserves int, baseFee int64, models *data.Models) (*accountSponsorshipService, error) {
```

Comment:
polishing: when we start adding lots of arguments to a method, it starts becoming hard to figure out what they're about. From a caller's perspective, it's hard to grasp what an int vs int64 means unless you take a deeper look into the docs.

I suggest using a struct to pass on these options.